packages/fxa-auth-db-mysql/lib/db/schema/patch-097-098.sql

```diff
@@ -0,0 +1,194 @@
+  id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  productId INT UNSIGNED NOT NULL,
+  capabilityId INT UNSIGNED NOT NULL,
+  UNIQUE INDEX subscriptionProductCapabilitiesUnique(productId, capabilityId)
+) ENGINE=InnoDB;
+
+CREATE TABLE IF NOT EXISTS subscriptionClientCapabilities (
+  id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  -- fxa_oauth.clients.id
+  clientId BINARY(8) NOT NULL,
+  capabilityId INT UNSIGNED NOT NULL,
+  UNIQUE INDEX subscriptionClientCapabilitiesUnique(clientId, capabilityId)
+) ENGINE=InnoDB;
+
+CREATE TABLE IF NOT EXISTS accountSubscriptionItems (
+  id INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  uid BINARY(16) NOT NULL,
+  productId INT UNSIGNED NOT NULL,
+  -- https://stripe.com/docs/api/customers/object#customer_object-id
+  subhubCustomerId VARCHAR(191),
+  -- https://stripe.com/docs/api/subscriptions/object#subscription_object-id
+  subhubSubscriptionId VARCHAR(191),
+  -- https://stripe.com/docs/api/subscription_items/object#subscription_item_object-id
+  subhubSubscriptionItemId VARCHAR(191),
+  createdAt BIGINT SIGNED NOT NULL,
+  FOREIGN KEY (uid) REFERENCES accounts(uid) ON DELETE CASCADE,
+  FOREIGN KEY (productId) REFERENCES subscriptionProducts(id) ON DELETE CASCADE,
+  INDEX accountSubscriptionsProductIdUid(productId, uid),
+  INDEX accountSubscriptionsUidSubscriptionId(uid, subhubSubscriptionId),
+  UNIQUE INDEX accountSubscriptionsUnique(uid, subhubSubscriptionId, subhubSubscriptionItemId)
+) ENGINE=InnoDB;
+
+CREATE PROCEDURE `fetchCustomerId_1` (
+  IN inUid BINARY(16)
+)
+BEGIN
+  SELECT subhubCustomerId
+  FROM accountSubscriptionItems
+  WHERE uid = inUid
+  LIMIT 1;
+END;
+
+CREATE PROCEDURE `createAccountSubscriptionItem_1` (
+  IN inUid BINARY(16),
+  IN inSubhubProductId VARCHAR(191),
+  IN inSubhubCustomerId VARCHAR(191),
+  IN inSubhubSubscriptionId VARCHAR(191),
+  IN inSubhubSubscriptionItemId VARCHAR(191),
+  IN inCreatedAt BIGINT SIGNED
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  SET @productId = NULL;
+  SELECT id INTO @productId FROM subscriptionProducts
+  WHERE
+    subhubProductId = inSubhubProductId;
+  
+  IF @productId IS NULL THEN
+    SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 2102,
+      MESSAGE_TEXT = 'Invalid product for subscription.';
+  END IF;
+
+  INSERT INTO accountSubscriptionItems(
+    uid,
+    productId,
+    subhubCustomerId,
+    subhubSubscriptionId,
+    subhubSubscriptionItemId,
+    createdAt
+  )
+  VALUES (
+    inUid,
+    @productId,
+    inSubhubCustomerId,
+    inSubhubSubscriptionId,
+    inSubhubSubscriptionItemId,
+    inCreatedAt
+  );
+END;
+
+CREATE PROCEDURE `cancelAccountSubscription_1` (
+  IN inUid BINARY(16),
+  IN inSubhubSubscriptionId VARCHAR(191)
+)
+BEGIN
+  DELETE FROM accountSubscriptionItems
+  WHERE
+    uid = inUid
+  AND
+    subhubSubscriptionId = inSubhubSubscriptionId;
```

Comment:
We should probably update the `deleteAccount` stored procedure to do the same in there too, right?

Reply:
I had assumed that the foreign key with on delete cascade would take care of that, but I'll take a look at deleteAccount

Reply:
I confess that I hadn't noticed your foreign key. That *would* take care of it, but I think we have operational concerns about foreign key constraints and are trying not to use them going forward. @jrgm can confirm or deny that...